### ISSUE TITLE
SentinelAPIError: HTTP status 200 OK: API response not valid. JSON decoding failed.

### ISSUE THREAD

**Issue by jpaleja** [sentinelsat/sentinelsat#415]
Hi,

I'm fairly new to python and was attempting to download data from sentinel within the boundaries of a geojson using this API. 

Heres the code:-

```python
import sentinelsat
import folium

from sentinelsat import SentinelAPI, read_geojson, geojson_to_wkt
from datetime import date

m = folium.Map([-18.349, 19.942], zoom_start=10)
boundsdata = r'C:\Users.......\Python\bounds_NAM.geo.json.json'
folium.GeoJson(boundsdata).add_to(m)`
m

user = '*****'
password = '******'
api = SentinelAPI(user, password, 'https://scihub.copernicus.eu/dhus')
footprint = geojson_to_wkt(read_geojson(boundsdata))
print (footprint)

products = api.query(footprint,
                     date = ('20190225', '20190227'),
                     platformname = 'Sentinel-2',
                     processinglevel = 'Level-2A',
                     cloudcoverpercentage = (0, 20))`
```

The following error appears:- 
```
KeyError                                  Traceback (most recent call last)
~\AppData\Roaming\Python\Python37\site-packages\sentinelsat\sentinel.py in _load_subquery(self, query, order_by, limit, offset)
    394             json_feed = response.json()["feed"]
--> 395             if json_feed["opensearch:totalResults"] is None:
    396                 # We are using some unintended behavior of the server that a null is

KeyError: 'opensearch:totalResults'

During handling of the above exception, another exception occurred:

SentinelAPIError                          Traceback (most recent call last)
<ipython-input-20-578efacc0a96> in <module>
      3                      platformname = 'Sentinel-2',
      4                      processinglevel = 'Level-2A',
----> 5                      cloudcoverpercentage = (0, 20))

~\AppData\Roaming\Python\Python37\site-packages\sentinelsat\sentinel.py in query(self, area, date, raw, area_relation, order_by, limit, offset, **keywords)
    206         )
    207         formatted_order_by = _format_order_by(order_by)
--> 208         response, count = self._load_query(query, formatted_order_by, limit, offset)
    209         self.logger.info("Found %s products", count)
    210         return _parse_opensearch_response(response)

~\AppData\Roaming\Python\Python37\site-packages\sentinelsat\sentinel.py in _load_query(self, query, order_by, limit, offset)
    347 
    348     def _load_query(self, query, order_by=None, limit=None, offset=0):
--> 349         products, count = self._load_subquery(query, order_by, limit, offset)
    350 
    351         # repeat query until all results have been loaded

~\AppData\Roaming\Python\Python37\site-packages\sentinelsat\sentinel.py in _load_subquery(self, query, order_by, limit, offset)
    401             total_results = int(json_feed["opensearch:totalResults"])
    402         except (ValueError, KeyError):
--> 403             raise SentinelAPIError("API response not valid. JSON decoding failed.", response)
    404 
    405         products = json_feed.get("entry", [])

SentinelAPIError: HTTP status 200 OK: API response not valid. JSON decoding failed.

```

Not sure what the issue is as i've followed the quickstart

Any help appreciated

Thanks

**Comment by j08lue** [comment]
Hmm. Never seen this one before. Can you please try with `https://scihub.copernicus.eu/apihub`, i.e. use the default? This is probably not the source of your error, but just for good measure, because `/dhus` is for the SciHub frontend and `/apihub` is for API access.

Could also be that the DHuS behavior changed. When you tested with `/apihub` and still get the same error, I can try to reproduce it and investigate.

**Comment by jpaleja** [comment]
Using `/apihub` i get a different error 

`KeyError                                  Traceback (most recent call last)`
`~\AppData\Roaming\Python\Python37\site-packages\sentinelsat\sentinel.py in _load_subquery(self, query, order_by, limit, offset)`
`    394             json_feed = response.json()["feed"]`
`--> 395             if json_feed["opensearch:totalResults"] is None:`
`    396                 # We are using some unintended behavior of the server that a null is`

`KeyError: 'opensearch:totalResults'`

`During handling of the above exception, another exception occurred:`

`SentinelAPIError                          Traceback (most recent call last)`
`<ipython-input-9-578efacc0a96> in <module>`
`      3                      platformname = 'Sentinel-2',`
`      4                      processinglevel = 'Level-2A',`
`----> 5                      cloudcoverpercentage = (0, 20))`

`~\AppData\Roaming\Python\Python37\site-packages\sentinelsat\sentinel.py in query(self, area, date, raw, area_relation, order_by, limit, offset, **keywords)`
`    206         )`
`    207         formatted_order_by = _format_order_by(order_by)`
`--> 208         response, count = self._load_query(query, formatted_order_by, limit, offset)`
`    209         self.logger.info("Found %s products", count)`
`    210         return _parse_opensearch_response(response)`

`~\AppData\Roaming\Python\Python37\site-packages\sentinelsat\sentinel.py in _load_query(self, query, order_by, limit, offset)`
`    347 `
`    348     def _load_query(self, query, order_by=None, limit=None, offset=0):`
`--> 349         products, count = self._load_subquery(query, order_by, limit, offset)`
`    350 `
`    351         # repeat query until all results have been loaded`

`~\AppData\Roaming\Python\Python37\site-packages\sentinelsat\sentinel.py in _load_subquery(self, query, order_by, limit, offset)`
 `   401             total_results = int(json_feed["opensearch:totalResults"])`
   ` 402         except (ValueError, KeyError):`
`--> 403             raise SentinelAPIError("API response not valid. JSON decoding failed.", response)`
`    404 `
`    405         products = json_feed.get("entry", [])`

`SentinelAPIError: HTTP status 200 OK: API response not valid. JSON decoding failed.`

Thanks

**Comment by JonasMedu** [comment]
got the same issue with
```python
footprint = "Intersects(POLYGON((-4.53 29.85, 26.75 29.85, 26.75 46.80,-4.53 46.80,-4.53 29.85)))"
products = api.query(footprint)
api.download_all(products)
```

`KeyError: 'opensearch:totalResults'`
...
`SentinelAPIError: HTTP status 200 OK: API response not valid. JSON decoding failed.`

where the poliygon: "Intersects(POLYGON((-4.53 29.85, 26.75 29.85, 26.75 46.80,-4.53 46.80,-4.53 29.85)))"
is the footprint of the example from [official docu](https://scihub.copernicus.eu/twiki/do/view/SciHubUserGuide/FullTextSearch?redirectedfrom=SciHubUserGuide.3FullTextSearch)
and returns a result if entered in https://scihub.copernicus.eu/dhus/#/home

**Comment by j08lue** [comment]
@JonasMedu `Insersects(POLYGON...)` is not a valid input Please only specify the WKT polygon string. The area relation `Intersects` is actually the default. If you wanted to change that, you could use the `area_relation` parameter to `api.query`, like `api.query(..., area_relation="Within")` or so.

Still, when you provide a bad area geometry, we should not fail with a `KeyError` but catch this better. Anyone wanting to submit a PR for this is very welcome. 😉

**Comment by valgur** [comment]
This issue seems to have been caused by a server-side issue that should now be fixed by #450.
